go/vt/binlog/keyrange_filter.go

```diff
@@ -40,40 +30,23 @@ func KeyRangeFilterFunc(kit key.KeyspaceIdType, keyrange *pb.KeyRange, sendReply
 				log.Warningf("Not forwarding DDL: %s", statement.Sql)
 				continue
 			case proto.BL_DML:
-				keyspaceIndex := strings.LastIndex(statement.Sql, KEYSPACE_ID_COMMENT)
-				if keyspaceIndex == -1 {
+				keyspaceId, unfriendly, err := annotations.ParseSQLAnnotation(string(statement.Sql))
+				if err != nil {
```

Comment:
If the action performed is different, then there's a justification for a separate return value.
I've also seen idiomatic uses where they check for a specific error to perform a different action. I think both approaches are reasonable.